Lib/test/test_contextlib.py

```diff
@@ -152,6 +152,28 @@ def woohoo():
         else:
             self.fail('StopIteration was suppressed')
 
+    def test_contextmanager_do_not_unchain_non_stopiteration_exceptions(self):
+        code = """\
+from __future__ import generator_stop
+from contextlib import contextmanager
+@contextmanager
+def test_issue29692():
+    try:
+        yield
+    except Exception as exc:
+        raise RuntimeError from exc
```

Comment:
Add some argument to the raised RuntimeError for checking that the correct RuntimeError is caught.